File: core/src/main/java/guri/kafka/blockchainchat/domain/Order.java
package guri.kafka.blockchainchat.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "order")
public class Order {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) @Column(name = "order_id")
    private Long id;
    @ManyToOne(targetEntity = User.class, fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(targetEntity = Product.class, fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private Product product;
    private int currentAmount;
    private int realizedAmount;
    private OrderType orderType;
    private LocalDateTime createdTime;


}

File: src/main/kotlin/guri/kafka/blockchainchat/order/Order.kt
package guri.kafka.blockchainchat.order

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Table("order")
class Order (
    @Id
    var id: Long? = null,
    var userId: Long,
    var productId: Long,
    var currentAmount: Int,
    var realizedAmount: Int,
    var createdTime : LocalDateTime,
    var orderType: OrderType
)